docs/checks.md

```diff
@@ -0,0 +1,299 @@
+
+
+**Example**
+```sql
+CREATE TABLE TABLE_NAME; -- False
+CREATE TABLE TableName; -- True
+CREATE TABLE table_name; -- False, line 3
+
+SELECT a FROM B WINDOW WindowName AS T; -- True
+SELECT a FROM B WINDOW window_name AS T; -- False, line 6
+
+CREATE TEMP TABLE FUNCTION BugScore(); -- True
+CREATE TEMP TABLE FUNCTION bugScore(); -- False
+CREATE TEMP TABLE FUNCTION bug_score(); -- false
+CREATE TEMP FUNCTION BugScore(); -- True, line 11
+
+CREATE TEMP FUNCTION A( s String ); -- False
+CREATE TEMP FUNCTION A( s STRING ); -- True
+CREATE TEMP FUNCTION A( s int64 ); -- False
+CREATE TEMP FUNCTION A( s INT64 ); -- True, line 16
+
+SELECT column_name; -- True
+SELECT TableName.column_name; -- True
+SELECT ColumnName; -- False
+SELECT COLUMN_NAME; -- False, line 21
+
+CREATE TEMPORARY FUNCTION A( string_param STRING ); -- True
+CREATE TEMPORARY FUNCTION A( STRING_PARAM STRING ); -- False
+CREATE TEMPORARY FUNCTION A( stringParam STRING ); -- False
+CREATE TEMPORARY FUNCTION A( StringParam STRING ); -- False, line 26
+
+CREATE PUBLIC CONSTANT TwoPi = 6.28; -- False
+CREATE PUBLIC CONSTANT TWO_PI = 6.28; -- True, line 29
+```
+
+**Linter Output**
+```
+In line 1, column 14: Table names or table aliases should be UpperCamelCase. [table-name]
+In line 3, column 14: Table names or table aliases should be UpperCamelCase. [table-name]
+In line 6, column 24: Window names should be UpperCamelCase. [window-name]
+In line 9, column 28: Function names should be UpperCamelCase. [function-name]
+In line 10, column 28: Function names should be UpperCamelCase. [function-name]
+In line 13, column 27: Simple SQL data types should be all caps. [data-type-name]
+In line 15, column 27: Simple SQL data types should be all caps. [data-type-name]
+In line 20, column 8: Column names should be lower_snake_case. [column-name]
+In line 21, column 8: Column names should be lower_snake_case. [column-name]
+In line 24, column 30: Function parameters should be lower_snake_case. [parameter-name]
+In line 25, column 30: Function parameters should be lower_snake_case. [parameter-name]
+In line 26, column 30: Function parameters should be lower_snake_case. [parameter-name]
+In line 28, column 24: Constant names should be CAPS_SNAKE_CASE. [constant-name]
+```
+
+## join
+Always explicitly indicate the type of join. Do not use just "JOIN", try to indicate the type like "INNER", "LEFT", etc.
+
+**Example**
+```sql
+-- Here is some valid JOIN usage.
+SELECT a FROM (t INNER JOIN x);
+SELECT a FROM (t LEFT JOIN x);
+SELECT a FROM (t RIGHT OUTER JOIN x);
+
+-- and also an invalid one.
+SELECT a FROM (t JOIN x); -- line 7
+
+```
+
+**Linter Output**
+```
+In line 7, column 18: Always explicitly indicate the type of join. [join]
+```
+
+## imports
+Imports should be on the top. 
+There shouldn't be any redundant(dublicate) imports.
+All 'PROTO' imports and all 'MODULE' imports should group among themselves, there shouldn't mixed order like (MODULE-PROTO-MODULE) or (PROTO-MODULE-PROTO)
```

Comment:
...should order as follows:
1. Proto imports
2. Module imports

or 

1. Module imports
2. Proto imports

Reply:
Fixed